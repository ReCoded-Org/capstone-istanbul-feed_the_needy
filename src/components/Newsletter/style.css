.newsletterContent {
  height: auto;
  width: 60em;
}

.ant-popover-inner-content button {
  display: inline-block;
  justify-content: center;
  align-items: center;
  margin: 3em;
  color: white;
  background-color: #e17007;
  padding: auto;
}

.popover {
  border: none;
  font-size: large;
  height: auto;
  margin-bottom: 2em;
  background-color: #e17007;
  color: white;
}

.popoverImage {
  width: 20em;
  height: 20em;
}

@media (max-width: 1000px) {
  .hidden-mobile {
    display: none;
  }
  .newsletterContent {
    height: auto;
    width: 10em;
  }
}
