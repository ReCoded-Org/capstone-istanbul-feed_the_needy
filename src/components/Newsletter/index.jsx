import React from "react";
import { Button, Row, Col, Popover } from "antd";
import MailchimpSubscribe from "react-mailchimp-subscribe";
import "./style.css";
import { useTranslation } from "react-i18next";
import heart from "./heart.png";

const Newsletter = (isTesting) => {
  const { t } = useTranslation();
  const content = (
    <div className="newsletterContent mc_embed_signup">
      <Row justify="space-around">
        <Col>
          <img src={heart} alt="heart" className="popoverImage hidden-mobile" />
        </Col>
        <Col>
          <h1>{t("newsletter.title0")}</h1>
          <br />
          <br />
          <MailchimpSubscribe
            url={process.env.REACT_APP_MAILCHIMP_URL}
            className="mailchimp"
          />
        </Col>
      </Row>
    </div>
  );

  return (
    <div justify="center" align="center">
      <Popover
        content={content}
        className="popover"
        title="Subscribe to our newsletter!"
      >
        {isTesting ? (
          <button type="button" className="myButton subscribeNewsletterButton">
            {t("newsletter.button0")}
          </button>
        ) : (
          <Button>{t("newsletter.button0")}</Button>
        )}
      </Popover>
    </div>
  );
};

export default Newsletter;
