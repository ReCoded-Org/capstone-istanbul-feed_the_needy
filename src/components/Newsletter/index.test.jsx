import React from "react";
import renderer from "react-test-renderer";
import Newsletter from "./index";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

// Fix for TypeError: window.matchMedia is not a function
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

it("Newsletter renders correctly", () => {
  const tree = renderer
    .create(<Newsletter isTesting />, { disableLifecycleMethods: true })
    .toJSON();
  expect(tree).toMatchSnapshot();
});
