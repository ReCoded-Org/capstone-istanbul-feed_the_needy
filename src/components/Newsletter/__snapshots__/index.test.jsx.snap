// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter renders correctly 1`] = `
<div
  align="center"
  justify="center"
>
  <button
    className="myButton subscribeNewsletterButton popover"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  >
    newsletter.button0
  </button>
</div>
`;
